### PR TITLE
refactor: [TRST-M-7] Incorrect permission check due to confusion between empty call and the zero selector

### PR DESCRIPTION
**Pull request by b00ste**
# What does this PR introduce?

## 🐛 Bug

Add an `isEmptyCall` parameter to `_extractExecuteParameters(..)` and retrieve it in `_extractCallType(..)`. If it is not true,
_ALLOWEDCALLS_WRITE should be turned on even for selector 0x00000000.

Fixes #???? 

### PR Checklist

- [x] Wrote Tests
- [ ] Wrote Documentation
- [x] Ran `npm run linter` (solhint)
- [x] Ran `npm run format` (prettier)
- [x] Ran `npm run build`
- [x] Ran `npm run test`

### CODE REVIEW

File: contracts/LSP6KeyManager/LSP6Modules/LSP6ExecuteModule.sol [hunk 1]
```diff
@@ -269,25 +270,22 @@ abstract contract LSP6ExecuteModule {
...
-                // CHECK if we are doing an empty call
-                (selector == bytes4(0) && value == 0) ||
-                // we do not require callType CALL
-                // if we are just transferring value without `data`
-                selector != bytes4(0)
-            ) {
-                requiredCallTypes = _ALLOWEDCALLS_WRITE;
-            }
+        if (operationType == OPERATION_0_CALL && !isEmptyCall) {
+            requiredCallTypes = _ALLOWEDCALLS_WRITE;
```
Comment (CJ42) [inline contracts/LSP6KeyManager/LSP6Modules/LSP6ExecuteModule.sol:279]: we should probably rename these `constant` to:

- `_ALLOWEDCALLS_WRITE ` --> `_ALLOWEDCALLS_CALL`
- `_ALLOWEDCALLS_READ` --> `_ALLOWEDCALLS_STATICCALL`
- `_ALLOWEDCALLS_EXECUTE` --> `_ALLOWEDCALLS_DELEGATECALL`
- `_ALLOWEDCALLS_VALUE` --> `_ALLOWEDCALLS_TRANSFERVALUE`

**Review by CJ42 — CHANGES_REQUESTED** [review]
we might need an extra test to check the fix work. Don't know which one tho

File: contracts/LSP6KeyManager/LSP6Modules/LSP6ExecuteModule.sol [hunk 2]
```diff
@@ -323,7 +322,7 @@ abstract contract LSP6ExecuteModule {
             ? bytes4(executeCalldata[164:168])
             : bytes4(0);
```
[the remark this answers is not in the document]
Comment (b00ste) [reply contracts/LSP6KeyManager/LSP6Modules/LSP6ExecuteModule.sol:323]: Yeah, that's for  getting the function selector, if you look a bit down we return `isEmptyCall = executeCalldata.length == 164`

**Review by CJ42 — APPROVED** [review]

File: contracts/Mocks/FallbackExtensions/GraffitiEventExtension.sol [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.4;
+
+/**
+ * @dev This contract is used only for testing purposes
+ */
+contract GraffitiEventExtension {
+    event GraffitiDataReceived(bytes graffitiData);
+
+    fallback(bytes calldata data) external payable returns (bytes memory) {
```
Comment (YamenMerhi) [inline contracts/Mocks/FallbackExtensions/GraffitiEventExtension.sol:10]: Can you resolve solhint warnings

**Review by YamenMerhi — APPROVED** [review]

File: contracts/Mocks/FallbackExtensions/GraffitiEventExtension.sol (hunk 3 above)
Comment (b00ste) [reply contracts/Mocks/FallbackExtensions/GraffitiEventExtension.sol:11]: done

**Pull request merged**